### PR TITLE
make attribute path private,can't be set

### PR DESCRIPTION
**Pull request by Harmouch101**
make attribute path private and removing both test_path_remove_prefix() and test_path_remove_prefix_no_prefix() in zenoh/test/test_path.py

### CODE REVIEW

**Comment by Harmouch101** [comment]
> Good! Now it's just missing the deletion of the remove_prefix() operation in path.py.
> Please remove it, and I'll accept your pull request.

@JEnoch fix it

File: zenoh/path.py [hunk 1]
```diff
@@ -18,7 +18,11 @@ def __init__(self, path):
         self.__path_regex = re.compile('^[^?#*]+$')
         if not self.is_valid(path):
             raise ValueError("Invalid Path: {}".format(path))
-        self.path = path
+        self._path = path
```
Comment (JEnoch) [inline zenoh/path.py]: Sorry but, I just realised you changed the attribute name "path" into "_path".
As a consequence, you should rename it all across this file (i.e. each "self.path" => "selaf._path").

Comment (Harmouch101) [reply zenoh/path.py]: @JEnoch done!

File: zenoh/path.py [hunk 2]
```diff
@@ -32,30 +36,26 @@ def is_valid(self, path):
...
+        return self._path
 
     def __len__(self):
-        return len(self.path)
+        return len(self._path)
 
     def __eq__(self, second_path):
         if isinstance(second_path, self.__class__):
-            return self.path == second_path.path
+            return self._path == second_path.path
```
Comment (JEnoch) [inline zenoh/path.py]:
```suggestion
            return self._path == second_path._path
```
You forgot this one ;)

Comment (JEnoch) [reply zenoh/path.py]: And also the one in \_\_hash\_\_() operation below.

Comment (Harmouch101) [reply zenoh/path.py]: > And also the one in __hash__() operation below.

@JEnoch done!

File: zenoh/path.py [hunk 3]
```diff
@@ -32,33 +36,29 @@ def is_valid(self, path):
...
+        return self._path
 
     def __len__(self):
-        return len(self.path)
+        return len(self._path)
 
     def __eq__(self, second_path):
         if isinstance(second_path, self.__class__):
-            return self.path == second_path.path
+            return self._path == second_path_.path
```
Comment (JEnoch) [inline zenoh/path.py]:
```suggestion
            return self._path == second_path._path
```
Typo: the _ should be added at the beginning of 'path' attribute. Not at the end of 'second_path'

File: zenoh/path.py [hunk 4]
```diff
@@ -32,33 +36,29 @@ def is_valid(self, path):
...
         return False
 
     def __str__(self):
-        return self.path
+        return self._ path
```
Comment (JEnoch) [inline zenoh/path.py]:
```suggestion
        return self._path
```
Typo: extra space character between '_' and 'path'.

File: zenoh/path.py (hunk 3 above)
Comment (Harmouch101) [reply zenoh/path.py]: @JEnoch 
what a stupid mistake hahaha
fixed it

File: zenoh/path.py (hunk 4 above)
Comment (Harmouch101) [reply zenoh/path.py]: @JEnoch 
fixed it !

**Pull request merged**